extensions/amp-animation/0.1/web-animations.js

```diff
@@ -891,6 +896,15 @@ class CssContextImpl {
 
     /** @private {?{width: number, height: number}} */
     this.viewportSize_ = null;
+
```

Comment:
IMHO we should just combine all these values into one `viewportParams` struct, initialize it to `null`, and resolve it once, e.g. via `getViewportParams_() {if (!viewportParams_) {...resolve...}}`. Otherwise it's hard to keep track which field available when: for us it's really they are either all available when resolved, or none are available.